superset-frontend/src/components/Menu/SubMenu.tsx

```diff
@@ -69,15 +68,24 @@ type MenuChild = {
   usesRouter?: boolean;
 };
 
+export enum ButtonStyleTypes {
+  Primary = 'primary',
+  Secondary = 'secondary',
+  Dashed = 'dashed',
+  Danger = 'danger',
+  Link = 'link',
+  Warning = 'warning',
+  Success = 'success',
+}
+
+interface ButtonProps {
+  name: any;
+  onClick: OnClickHandler;
+  style: ButtonStyleTypes;
```

Comment:
Curious what you think about skipping the enum/export and just doing:

```suggestion
  style: 'primary' | 'secondary' | 'tertiary' | 'dashed' | 'danger' | 'link' | 'warning' | 'success';
```

It seems a little simpler/cleaner, since:
* You don't have to export anything here
* You won't have to import ButtonStyleTypes when you adda button, you can use the string directly (e.g. `style: 'primary',`) as Typescript will know the valid values anyway

Reply:
I tried doing this but it didn't like the assignment when doing `style:< whatever defined string>` in the button. I'll play around with it more and see what i'm missing...